cpp/src/orderby/is_sorted.cu

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/cudf.h>
+#include <utilities/cudf_utils.h>
+#include <utilities/error_utils.hpp>
+
+#include <table/device_table.cuh>
+#include <table/device_table_row_operators.cuh>
+
+#include <rmm/thrust_rmm_allocator.h>
+
+/* ----------------------------------------------------------------------------*/
+/**
+ * @brief Wrapper around row_inequality_comparator to pass row and row+1
+ */
+/* ----------------------------------------------------------------------------*/
+
+template <bool nullable = true>
+struct compare_with_neighbor{
+
+    compare_with_neighbor(device_table t, bool nulls_are_smallest, int8_t *const asc_desc_flags = nullptr) 
+        : compare(t,t, nulls_are_smallest, asc_desc_flags) {}
+    
+    __device__ bool operator()(gdf_index_type i){
+       return compare(i, i + 1);
+    }
+    row_inequality_comparator<nullable> compare;
+};
+
+/* --------------------------------------------------------------------------*/
+/** 
+ * @brief Checks if a Table is sorted along the row.
+ * 
+ * @param[in] input_table 	 whose rows needs to be validated 
+ * @param[in] ascending          vector representing the relation between columns
+ * 				 (0 is ascending order and 1 is descending)
+ * @param[in] nulls_are_smallest Flag to indicate if nulls are to be considered
+ * 				 smaller than non-nulls or viceversa
+ * 
+ * @returns true - if sorted , false - if not.
+ */
+/* ----------------------------------------------------------------------------*/
+bool is_sorted(cudf::table const &table,
+                       std::vector<int8_t>& ascending,
+                       bool nulls_are_smallest = false)                       
+{
+  cudaStream_t stream = 0;
+  bool sorted = false;
+
+  if (static_cast <unsigned int>(table.num_columns()) != ascending.size())
+  {
+      return false;
+  }
+  else if (table.num_columns() == 0)
```

Comment:
@kkraus14 what's the expected behavior for empty inputs? Should they return `true` or `false` or throw an exception?

Reply:
> @kkraus14 what's the expected behavior for empty inputs? Should they return `true` or `false` or throw an exception?

Pandas behavior is to return True but I'm fine with whatever you want to do from a C++ perspective.

Reply:
@jrhemstad @kkraus14 A user might just provide an empty input unknowingly and that might be an issue  he he relies on that. So, it would be better with an exception as at least it will point the issue.

Reply:
The problem is that Dask uses empty inputs all the time, and we can't throw an error for those cases.

Reply:
So, shall I go with behavior or pandas and return True ?

Reply:
We can happily Try / Except from the Python side or special case 0 sized inputs for these functions from the Python side as well. As long as we don't segfault we should be okay :)

Reply:
Currently returning true, if there are any comments on it, I will change it later.